pkg/auth/providers/oidc/oidc_provider.go

```diff
@@ -515,3 +531,36 @@ func (o *OpenIDCProvider) getOIDCProvider(ctx context.Context, oidcConfig *v32.O
 	// This will perform discovery in the oidc library
 	return oidc.NewProvider(ctx, oidcConfig.Issuer)
 }
+
+func isValidACR(claimACR string, configuredACR string) bool {
+	// if we have no ACR configured, all values are accepted
+	if configuredACR == "" {
+		return true
+	}
+
+	if claimACR != configuredACR {
+		logrus.Infof("acr value in token does not match configured acr value")
+		return false
+	}
+	return true
+}
+
+func parseACRFromAccessToken(accessToken string) (string, error) {
+	tokenClaims := make(map[string]interface{})
+	segments := strings.Split(accessToken, ".")
+	if len(segments) < 2 {
+		return "", errors.New("invalid access token format")
+	}
+	decodedClaims, err := base64.URLEncoding.DecodeString(segments[1])
+	if err != nil {
+		return "", fmt.Errorf("decoding token claims: %w", err)
+	}
+	if err := json.Unmarshal(decodedClaims, &tokenClaims); err != nil {
+		return "", fmt.Errorf("parsing token claims: %w", err)
+	}
+	acrValue, ok := tokenClaims["acr"].(string)
+	if !ok {
+		return "", errors.New("acr claim not found in token")
+	}
+	return acrValue, nil
+}
```

Comment:
This manual decoding of the payload fails for some inputs, because of the `base64.URLEncoding`. You should use the `base64.RawURLEncoding` or  `base64.RawStdEncoding` instead, that are not expecting paddings in the encoded strings.

You can try with this random token:

```go
const accessToken = `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCJhY3IiOiJmb29iYXIifQ.kKvbaR4j2N7FV1PWI1uH4YphC0KuL8YVsXDCzwcbSsg`
```

https://play.golang.com/p/z38g26yyYiB

If we are not going to use the returned verified token from the `_, err = verifier.Verify(updatedContext, oauth2Token.AccessToken)` at line 391 as suggested, and extracting manually the claims, then we can stick using the `ParseUnverified` of the jwt library. It's probably cleaner and safer:

```suggestion
func parseACRFromAccessToken(accessToken string) (string, error) {
	var parser jwt.Parser
	// we already validated the incoming token
	token, _, err := parser.ParseUnverified(accessToken, jwt.MapClaims{})
	if err != nil {
		return "", fmt.Errorf("failed to parse token: %w", err)
	}

	claims, ok := token.Claims.(jwt.MapClaims)
	if !ok {
		return "", errors.New("invalid access token jwt.MapClaims format")
	}

	acrValue, found := claims["acr"].(string)
	if !found {
		return "", fmt.Errorf("acr claim invalid or not found in token: (acr=%v)", claims["acr"])
	}
	return acrValue, nil
}
```

https://play.golang.com/p/-2qFOF1IndC